mono/sgen/sgen-marksweep.c

```diff
@@ -1787,6 +1789,72 @@ major_finish_nursery_collection (void)
 #endif
 }
 
+static int
+block_usage_comparer (const void *bl1, const void *bl2)
+{
```

Comment:
You can just return `nused2 - nused1` if you cast both to appropriate signed types.